controller/rest/file_monitor.go

```diff
@@ -227,7 +238,7 @@ func handlerFileMonitorConfig(w http.ResponseWriter, r *http.Request, ps httprou
 			}
 
 			for i, cfilter := range profConf.Filters {
-				if cfilter.Filter == filter.Filter {
+				if (cfilter.Filter == filter.Filter) || (cfilter.Path == base) {
```

Comment:
Note that `Filter` is not the path, this is the pseudo regex expression. eg: /tmp/.*

While the additional check if Path is the same as base will throw an error if the path is the same. The reason is that the monitoring logic can only store one filter per directory.

Reply:
Please add some test cases and make sure this logic does help.